### PR TITLE
Fix issue where non-string heartbeat details and only the first heartbeat detail shows up in CLI

### PR DESCRIPTION
**Pull request by mastermanu**
The CLI had two bugs:
1) If the heartbeat detail written was not a string, the CLI would crash as it would attempt to cast whatever value type the heartbeat detail was to a string.
2) If the client was recording multiple heartbeat details (which is possible as RecordHeartbeat takes in a Variadic), it was only displaying the first one.

This change fixes both issues. It was verified on a local development environment. Here is example output for the following record heartbeat call:

```
		x := []string{"1", "2", "3"}
		y := []byte{1, 8, 0, 3}
		time.Sleep(7 * time.Second)
		activity.RecordHeartbeat(d.ctx, i, "my_name_is_manu", batchDetails{Count: i, Name: "Manu"}, x, y)
```

```
  "pendingActivities": [
    {
      "activityId": "5",
      "activityType": {
        "name": "batch-driver-activity"
      },
      "state": "Started",
      "heartbeatDetails": "[0, my_name_is_manu, map[Count:0 Name:Manu], [1 2 3], AQgAAw]",
      "lastHeartbeatTimestamp": "2020-07-09T22:21:35-07:00",
      "lastStartedTimestamp": "2020-07-09T22:21:28-07:00",
      "scheduledTimestamp": "1969-12-31T16:00:00-08:00",
      "expirationTimestamp": "2020-07-12T22:21:28-07:00",
      "lastFailure": "nil"
    }
  ]

```
There are no unit tests that cover retrieving heartbeat details for the CLI at all. These need to be added in a future commit. Unit tests for CLI are in general quite bad (less than 22% code coverage), so a separate effort needs to be kicked off for that.

There is no virtually risk to this change.

### CODE REVIEW

File: proto/internal/temporal/server/api/cli/v1/message.proto [hunk 1]
```diff
@@ -54,7 +54,7 @@ message PendingActivityInfo {
     string activity_id = 1;
     temporal.api.common.v1.ActivityType activity_type = 2;
     temporal.api.enums.v1.PendingActivityState state = 3;
-    string heartbeat_details = 4;
+    repeated string heartbeat_details = 4;
```
Comment (samarabbas) [inline proto/internal/temporal/server/api/cli/v1/message.proto:57]: I'm actually confused why heartbeat details are modeled as string.  I think this should be blob like we model where user Payload(HHeader, Data) is passed through the system.  @alexshtin can you confirm if this is something we missed and needs to be converted to Payload type.

Comment (mastermanu) [reply proto/internal/temporal/server/api/cli/v1/message.proto:57]: yeah, i can plumb the encoding and the payload directly through as well. good point on questioning the assumption on this being a string

Comment (mastermanu) [reply proto/internal/temporal/server/api/cli/v1/message.proto:57]: the recordheartbeat and getheartbeat activity functions do take in whole objects. the "header" is part of our internal de/serialization mechanism where we decide whether to write a raw byte array directly vs. convert the object to json. so I can understand why we wouldn't want to at least surface the header part out of the CLI.

This change breaks if we try to record a raw byte array though, but I think getting raw byte array details out of the CLI has been broken for a while

Comment (mastermanu) [reply proto/internal/temporal/server/api/cli/v1/message.proto]: moved bulk of this logic into SDK. Diff is smaller now.

**Pull request merged**